lib/protobuf/nats/client.rb

```diff
@@ -13,8 +14,85 @@ def close_connection
         # no-op (I think for now), the connection to server is persistent
       end
 
+      def self.subscription_key_cache
+        @subscription_key_cache ||= {}
+      end
+
       def send_request
-        # after method execution the "response" should be in @response_data and should be a string of bytes
+        retries ||= 3
+
+        setup_connection
```

Comment:
This sets up some stats and things. Idk if we need it, but the other connectors are using this: https://github.com/ruby-protobuf/protobuf/blob/master/lib/protobuf/rpc/connectors/base.rb#L159